src/wristpy/core/computations.py

```diff
@@ -43,7 +43,54 @@ def moving_mean(array: Measurement, epoch_length: int = 5) -> Measurement:
     if array.measurements.ndim == 1:
         measurements_mean_array = measurements_mean_array.flatten()
 
-    return Measurement(
+    return models.Measurement(
         measurements=measurements_mean_array,
         time=measurement_df_mean["time"],
     )
+
+
+def moving_median(
+    acceleration: models.Measurement, window_size: int
+) -> models.Measurement:
+    """Applies moving median to acceleration data.
+
+    Args:
+        acceleration: the three dimensional accelerometer data. A Measurement object,
+        it will have two attributes. 1) measurements, containing the three dimensional
+        accelerometer data in an np.array and 2) time, a pl.Series containing
+        datetime.datetime objects.
+
+        window_size: The size of the overlapping window within which the median will be
+        applied. Rolling median window is centered. Measured in seconds.
+
+
+    Returns:
+        Measurement object with rolling median applied to the measurement data. The
+        measurements data will retain it's shape, and the time data will be returned
+        unaltered.
+    """
+    measurements_polars_df = pl.concat(
+        [pl.DataFrame(acceleration.measurements), pl.DataFrame(acceleration.time)],
+        how="horizontal",
+    )
+    measurements_polars_df = measurements_polars_df.set_sorted("time")
+    offset = -((window_size // 2) + 1)
+    offset_str = str(offset) + "s"
+    moving_median_df = measurements_polars_df.select(
```

Comment:
This function will not be able to take the moving median of anything but acceleration as it will fail on data that isn't N-by-3. Are we sure that we never need this for any other measurement? If no, can we just copy the implementation of `moving_mean` as I believe this one is valid for any N-by-M array.

Reply:
That's true I think this is easy to fix by using "*" instead of naming the cols. Then it will work on any amount of columns that comes with a timestamp col. 



here's the revised portion: 

 moving_median_df = measurements_polars_df.select(
        [
            pl.median("*").rolling(
                index_column="time", period=f"{window_size}s", offset=offset_str
            )
        ]
    )

    return models.Measurement(
        measurements=moving_median_df.drop("time").to_numpy(),
        time=measurements_polars_df["time"],
    )

Reply:
That's true I can edit this to cover any amount of cols like so: 

 ```
moving_median_df = measurements_polars_df.select(
        [
            pl.median("*").rolling(
                index_column="time", period=f"{window_size}s", offset=offset_str
            )
        ]
    )

    return models.Measurement(
        measurements=moving_median_df.drop("time").to_numpy(),
        time=measurements_polars_df["time"],
    )
    
    
```